Docker/JobDefinition.php

```diff
@@ -0,0 +1,218 @@
+<?php
+namespace Keboola\DockerBundle\Docker;
+
+use Keboola\Syrup\Exception\UserException;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+
+class JobDefinition
+{
+    /**
+     * @var
+     */
+    private $configId;
+
+    /**
+     * @var string
+     */
+    private $rowId;
+
+    /**
+     * @var string
+     */
+    private $configVersion;
+
+    /**
+     * @var string
+     */
+    private $rowVersion;
+
+    /**
+     * @var Component
+     */
+    private $component;
+
+    /**
+     * @var array
+     */
+    private $configuration = [];
+
+    /**
+     * @var array
+     */
+    private $state = [];
+
+    /**
+     * @var bool
+     */
+    private $isDisabled = false;
+
+    public function __construct($configuration)
+    {
+        $this->configuration = $this->normalizeConfiguration($configuration);
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getComponentId()
+    {
+        return $this->component ? $this->component->getId() : null;
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getConfigId()
+    {
+        return $this->configId;
+    }
+
+    /**
+     * @param mixed $configId
+     * @return $this
+     */
+    public function setConfigId($configId)
+    {
+        $this->configId = $configId;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getRowId()
+    {
+        return $this->rowId;
+    }
+
+    /**
+     * @param string $rowId
+     * @return $this
+     */
+    public function setRowId($rowId)
+    {
+        $this->rowId = $rowId;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getConfigVersion()
+    {
+        return $this->configVersion;
```

Comment:
prijde mi, ze se to configVersion na nic nepozuiva? je to k necemu nebo jen pro strycka prihodu?

Reply:
viz https://github.com/keboola/docker-bundle/pull/196#discussion_r144222886

Reply:
používá se tu - https://github.com/keboola/docker-bundle/pull/196/files#diff-9eb2f23f24e76893b04f474cd5a56586R174

(Runner.php je v diffu defaultně skrytej, protože prej moc velkej diff)

Reply:
jasne, mel jsem pocit, ze jsem to nekde videl!